libnetdata/socket/socket.c

```diff
@@ -808,9 +879,17 @@ int connect_to_this(const char *definition, int default_port, struct timeval *ti
 
     debug(D_CONNECT_TO, "Attempting connection to host = '%s', service = '%s', interface = '%s', protocol = %d (tcp = %d, udp = %d)", host, service, interface, protocol, IPPROTO_TCP, IPPROTO_UDP);
 
-    if(!*host) {
-        error("Definition '%s' does not specify a host.", definition);
-        return -1;
+    int test = test_tcp_hostname(host);
```

Comment:
The `host` still points to the beginning of the string, not the part that contains the hostname.

I tested it with:
```
    destination = tcp://mechani.se:19999
```
And the error that I get in the logs is:
```
2020-01-24 09:33:32: netdata ERROR : STREAM_SENDER[loki] : Definition 'tcp:' does not specify a host.
```

Reply:
The last commit brought the message that has been parsed instead of the message that was parsed.